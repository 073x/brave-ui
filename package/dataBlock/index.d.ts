/// <reference types="react" />
import * as React from 'react';
export interface DataProps {
    id?: string;
    size?: 'medium' | 'small';
    list?: boolean;
    children?: React.ReactNode;
}
declare class DataBlock extends React.PureComponent<DataProps, {}> {
    render(): JSX.Element;
}
export interface DataItemProps {
    id?: string;
    color?: string;
    counter?: string;
    text?: string;
    description?: string;
    size?: 'medium' | 'small';
    onClick?: (e: any) => void;
    noSelect?: boolean;
}
declare class DataItem extends React.PureComponent<DataItemProps, {}> {
    render(): JSX.Element;
}
export { DataBlock, DataItem };
