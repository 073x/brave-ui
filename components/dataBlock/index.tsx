import * as React from 'react'

import {
  StyledDataBlock,
  StyledDataItem,
  StyledDataItemCounter,
  StyledDataItemText,
  StyledDataItemDescription
} from './style'

export interface DataProps {
  id?: string,
  list?: boolean,
  children?: React.ReactNode
}

class DataBlock extends React.PureComponent<DataProps, {}> {
  render () {
    const { id, list, children } = this.props
    return (
      <StyledDataBlock id={id} list={list}>{children}</StyledDataBlock>
    )
  }
}
export interface DataItemTheme {
  color?: string,
  userSelect?: 'auto' | 'none'
}

export interface DataItemProps {
  id?: string,
  counter?: string | number,
  text?: string,
  description?: string,
  size?: 'medium' | 'small',
  onClick?: (e: any) => void,
  theme?: {
    [dataItem: string]: DataItemTheme
  }
}

class DataItem extends React.PureComponent<DataItemProps, {}> {
  render () {
    const {
      id,
      theme,
      counter,
      text,
      description,
      onClick,
      size = 'medium'
    } = this.props

    return (
      <StyledDataItem id={id} theme={theme} onClick={onClick} size={size}>
        <StyledDataItemCounter theme={theme} size={size}>{counter}</StyledDataItemCounter>
          {
            text && <StyledDataItemText theme={theme} size={size}>{text}</StyledDataItemText>
          }
          {
          description && <StyledDataItemDescription size={size}>{description}</StyledDataItemDescription>
          }
      </StyledDataItem>
    )
  }
}

export {
  DataBlock,
  DataItem
}
