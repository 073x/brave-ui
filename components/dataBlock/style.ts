/* This Source Code Form is subject to the terms of the Mozilla Public
 * License. v. 2.0. If a copy of the MPL was not distributed with this file.
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import styled from 'styled-components'
import { DataItemProps } from './index'

// s/Data/data

const StyledDataBlock = styled.ul`
  box-sizing: border-box;
  display: inline-block;
  font-weight: 400;
  margin: 0;
  padding: 0;
` as any

const StyledDataItem = styled.li`
  box-sizing: border-box;
  margin-right: 40px;
  margin-bottom: 20px;
  display: inline-block;
  vertical-align: middle;
  list-style-type: none;
` as any

const StyledDataItemCounter = styled.div`
  box-sizing: border-box;
  color: ${(p: DataItemProps) => p.color || 'inherit'}
  letter-spacing: -0.4px;
  font-size: 44px;
  line-height: 53px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  max-width: 200px;
` as any

const StyledDataItemText = styled.span`
  box-sizing: border-box;
  display: inline;
  color: #999999;
  font-size: 20px;
  letter-spacing: 0;
  line-height: 24px;
  margin-left: 3px;
` as any

const StyledDataItemDescription = styled.div`
  box-sizing: border-box;
  font-size: 13px;
  color: inherit;
  margin-left: 3px;
` as any

export {
  StyledDataBlock,
  StyledDataItem,
  StyledDataItemCounter,
  StyledDataItemText,
  StyledDataItemDescription
}
