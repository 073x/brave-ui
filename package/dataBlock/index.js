"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const style_1 = require("./style");
class DataBlock extends React.PureComponent {
    render() {
        const { id, list, children } = this.props;
        return (React.createElement(style_1.StyledDataBlock, { id: id, list: list }, children));
    }
}
exports.DataBlock = DataBlock;
class DataItem extends React.PureComponent {
    render() {
        const { id, color, counter, text, description, onClick, noSelect, size = 'medium' } = this.props;
        return (React.createElement(style_1.StyledDataItem, { id: id, color: color, onClick: onClick, noSelect: noSelect, size: size },
            React.createElement(style_1.StyledDataItemCounter, { color: color, size: size }, counter),
            text && React.createElement(style_1.StyledDataItemText, { size: size }, text),
            description && React.createElement(style_1.StyledDataItemDescription, { size: size }, description)));
    }
}
exports.DataItem = DataItem;
//# sourceMappingURL=index.js.map