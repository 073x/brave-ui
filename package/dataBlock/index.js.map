{"version":3,"file":"index.js","sourceRoot":"","sources":["../../components/dataBlock/index.tsx"],"names":[],"mappings":";;AAAA,+BAA8B;AAE9B,mCAMgB;AAQhB,eAAgB,SAAQ,KAAK,CAAC,aAA4B;IACxD,MAAM;QACJ,MAAM,EAAE,EAAE,EAAE,IAAI,EAAE,QAAQ,EAAE,GAAG,IAAI,CAAC,KAAK,CAAA;QACzC,OAAO,CACL,oBAAC,uBAAe,IAAC,EAAE,EAAE,EAAE,EAAE,IAAI,EAAE,IAAI,IAAG,QAAQ,CAAmB,CAClE,CAAA;IACH,CAAC;CACF;AA+CC,8BAAS;AAlCX,cAAe,SAAQ,KAAK,CAAC,aAAgC;IAC3D,MAAM;QACJ,MAAM,EACJ,EAAE,EACF,KAAK,EACL,OAAO,EACP,IAAI,EACJ,WAAW,EACX,OAAO,EACP,QAAQ,EACR,IAAI,GAAG,QAAQ,EAChB,GAAG,IAAI,CAAC,KAAK,CAAA;QAEd,OAAO,CACL,oBAAC,sBAAc,IACb,EAAE,EAAE,EAAE,EACN,KAAK,EAAE,KAAK,EACZ,OAAO,EAAE,OAAO,EAChB,QAAQ,EAAE,QAAQ,EAClB,IAAI,EAAE,IAAI;YAEV,oBAAC,6BAAqB,IAAC,KAAK,EAAE,KAAK,EAAE,IAAI,EAAE,IAAI,IAAG,OAAO,CAAyB;YAE9E,IAAI,IAAI,oBAAC,0BAAkB,IAAC,IAAI,EAAE,IAAI,IAAG,IAAI,CAAsB;YAGrE,WAAW,IAAI,oBAAC,iCAAyB,IAAC,IAAI,EAAE,IAAI,IAAG,WAAW,CAA6B,CAElF,CAClB,CAAA;IACH,CAAC;CACF;AAIC,4BAAQ"}