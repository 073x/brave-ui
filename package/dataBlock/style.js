"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const styled_components_1 = require("styled-components");
const helpers_1 = require("../helpers");
const StyledDataBlock = styled_components_1.default.ul `
  box-sizing: border-box;
  display: ${(p) => !p.list ? 'inline-block' : 'block'};
  -webkit-font-smoothing: antialiased;
  font-weight: 400;
  margin: 0;
  padding: 0;
  color: inherit;
  font-size: inherit;
  font-family: inherit;

  & > li {
    display: ${(p) => p.list ? 'flex' : 'inline-block'};
    align-items: ${(p) => p.list ? 'center' : null};
    min-height: ${(p) => p.list ? '30px' : null};
    margin-right: ${(p) => p.list ? '0' : '40px'};
    margin-bottom: ${(p) => p.list ? '0' : '20px'};
  }
`;
exports.StyledDataBlock = StyledDataBlock;
const StyledDataItem = styled_components_1.default.li `
  user-select: ${(p) => p.noSelect ? 'none' : 'auto'};
  box-sizing: border-box;
  vertical-align: middle;
  list-style-type: none;
  color: inherit;
  font-size: inherit;
  font-family: inherit;
`;
exports.StyledDataItem = StyledDataItem;
const StyledDataItemCounter = styled_components_1.default.span `
  color: ${(p) => p.color || 'inherit'}
  font-size: ${(p) => helpers_1.setSize(p, null, '44px', '26px')};
  line-height: ${(p) => helpers_1.setSize(p, null, '52px', '24px')};
  max-width: ${(p) => helpers_1.setSize(p, null, '200px', '38px')};
  text-align: ${(p) => p.size === 'small' ? 'right' : 'inherit'};
  box-sizing: border-box;
  letter-spacing: -0.4px;
  font-family: inherit;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
`;
exports.StyledDataItemCounter = StyledDataItemCounter;
const StyledDataItemText = styled_components_1.default.span `
  font-size: ${(p) => helpers_1.setSize(p, null, '20px', '13px')};
  line-height: ${(p) => helpers_1.setSize(p, null, '24px', '16px')};
  margin-left: ${(p) => helpers_1.setSize(p, null, '3px', '10px')};
  box-sizing: border-box;
  display: inline;
  color: inherit;
  font-family: inherit;
  letter-spacing: 0;
`;
exports.StyledDataItemText = StyledDataItemText;
const StyledDataItemDescription = styled_components_1.default.div `
  box-sizing: border-box;
  font-size: 13px;
  line-height: ${(p) => helpers_1.setSize(p, null, '24px', '16px')};
  font-family: inherit;
  color: inherit;
  margin-left: 3px;
`;
exports.StyledDataItemDescription = StyledDataItemDescription;
//# sourceMappingURL=style.js.map