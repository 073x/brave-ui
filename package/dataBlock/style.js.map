{"version":3,"file":"style.js","sourceRoot":"","sources":["../../components/dataBlock/style.ts"],"names":[],"mappings":";;AAIA,yDAAsC;AAEtC,wCAAgD;AAIhD,MAAM,eAAe,GAAG,2BAAM,CAAC,EAAE,CAAA;;aAEpB,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,CAAC,IAAI,CAAC,CAAC,CAAC,cAAc,CAAC,CAAC,CAAC,OAAO;;;;;;;;;;eAUlD,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,IAAI,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,cAAc;mBAC9C,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,IAAI,CAAC,CAAC,CAAC,QAAQ,CAAC,CAAC,CAAC,IAAI;kBAC3C,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,IAAI,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,IAAI;oBACtC,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,IAAI,CAAC,CAAC,CAAC,GAAG,CAAC,CAAC,CAAC,MAAM;qBACtC,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,IAAI,CAAC,CAAC,CAAC,GAAG,CAAC,CAAC,CAAC,MAAM;;CAEpD,CAAA;AA+CN,0CAAe;AA7CjB,MAAM,cAAc,GAAG,2BAAM,CAAC,EAAE,CAAA;iBACf,CAAC,CAAgB,EAAE,EAAE,CAAC,CAAC,CAAC,QAAQ,CAAC,CAAC,CAAC,MAAM,CAAC,CAAC,CAAC,MAAM;;;;;;;CAO3D,CAAA;AAsCN,wCAAc;AApChB,MAAM,qBAAqB,GAAG,2BAAM,CAAC,IAAI,CAAA;WAC9B,CAAC,CAAgB,EAAE,EAAE,CAAC,CAAC,CAAC,KAAK,IAAI,SAAS;eACtC,CAAC,CAAgB,EAAE,EAAE,CAAC,6BAAmB,CAAC,CAAC,CAAC,IAAI,EAAE,MAAM,EAAE,MAAM,EAAE,SAAS,CAAC;iBAC1E,CAAC,CAAgB,EAAE,EAAE,CAAC,6BAAmB,CAAC,CAAC,CAAC,IAAI,EAAE,MAAM,EAAE,MAAM,EAAE,SAAS,CAAC;eAC9E,CAAC,CAAgB,EAAE,EAAE,CAAC,6BAAmB,CAAC,CAAC,CAAC,IAAI,EAAE,OAAO,EAAE,MAAM,EAAE,SAAS,CAAC;gBAC5E,CAAC,CAAgB,EAAE,EAAE,CAAC,CAAC,CAAC,IAAI,KAAK,OAAO,CAAC,CAAC,CAAC,OAAO,CAAC,CAAC,CAAC,SAAS;;;;;;;CAOtE,CAAA;AAyBN,sDAAqB;AAvBvB,MAAM,kBAAkB,GAAG,2BAAM,CAAC,IAAI,CAAA;WAC3B,CAAC,CAAgB,EAAE,EAAE,CAAC,CAAC,CAAC,KAAK,IAAI,SAAS;eACtC,CAAC,CAAgB,EAAE,EAAE,CAAC,6BAAmB,CAAC,CAAC,CAAC,IAAI,EAAE,MAAM,EAAE,MAAM,EAAE,SAAS,CAAC;iBAC1E,CAAC,CAAgB,EAAE,EAAE,CAAC,6BAAmB,CAAC,CAAC,CAAC,IAAI,EAAE,MAAM,EAAE,MAAM,EAAE,SAAS,CAAC;iBAC5E,CAAC,CAAgB,EAAE,EAAE,CAAC,6BAAmB,CAAC,CAAC,CAAC,IAAI,EAAE,KAAK,EAAE,MAAM,EAAE,SAAS,CAAC;;;;;CAKpF,CAAA;AAeN,gDAAkB;AAbpB,MAAM,yBAAyB,GAAG,2BAAM,CAAC,GAAG,CAAA;;;iBAG3B,CAAC,CAAgB,EAAE,EAAE,CAAC,6BAAmB,CAAC,CAAC,CAAC,IAAI,EAAE,MAAM,EAAE,MAAM,EAAE,SAAS,CAAC;;;;CAIrF,CAAA;AAON,8DAAyB"}