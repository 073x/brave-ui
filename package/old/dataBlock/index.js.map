{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/old/dataBlock/index.tsx"],"names":[],"mappings":";;AAAA,+BAA8B;AAE9B,mCAMgB;AAQhB,eAAgB,SAAQ,KAAK,CAAC,aAA4B;IACxD,MAAM;QACJ,MAAM,EAAE,EAAE,EAAE,MAAM,EAAE,QAAQ,EAAE,GAAG,IAAI,CAAC,KAAK,CAAA;QAC3C,OAAO,CACL,oBAAC,uBAAe,IAAC,EAAE,EAAE,EAAE,EAAE,MAAM,EAAE,MAAM,IAAG,QAAQ,CAAmB,CACtE,CAAA;IACH,CAAC;CACF;AAgDC,8BAAS;AA/BX,cAAe,SAAQ,KAAK,CAAC,aAAgC;IAC3D,MAAM;QACJ,MAAM,EACJ,EAAE,EACF,KAAK,EACL,OAAO,EACP,IAAI,EACJ,WAAW,EACX,OAAO,EACP,IAAI,GAAG,QAAQ,EAChB,GAAG,IAAI,CAAC,KAAK,CAAA;QAEd,OAAO,CACL,oBAAC,sBAAc,IAAC,EAAE,EAAE,EAAE,EAAE,KAAK,EAAE,KAAK,EAAE,OAAO,EAAE,OAAO,EAAE,IAAI,EAAE,IAAI;YAChE,oBAAC,6BAAqB,IAAC,KAAK,EAAE,KAAK,EAAE,IAAI,EAAE,IAAI,IAAG,OAAO,CAAyB;YAE9E,IAAI;gBACJ,CAAC,CAAC,oBAAC,0BAAkB,IAAC,KAAK,EAAE,KAAK,EAAE,IAAI,EAAE,IAAI,IAAG,IAAI,CAAsB;gBAC3E,CAAC,CAAC,IAAI;YAGN,WAAW;gBACX,CAAC,CAAC,oBAAC,iCAAyB,IAAC,KAAK,EAAE,KAAK,EAAE,IAAI,EAAE,IAAI,IAAG,WAAW,CAA6B;gBAChG,CAAC,CAAC,IAAI,CAEK,CAClB,CAAA;IACH,CAAC;CACF;AAIC,4BAAQ"}