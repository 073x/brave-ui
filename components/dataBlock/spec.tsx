/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import { DataBlock, DataItem } from '../dataBlock'

describe('dataBlock tests', () => {
  const baseDataBlockComponent = (props?: object) => <DataBlock {...props} />
  const baseDataItemComponent = (props?: object) => <DataItem {...props} />

  describe('dataBlock component tests', () => {
    it('matches the snapshot', () => {
      const component = baseDataBlockComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseDataBlockComponent({id: 'toughDataBlock'}))
      const assertion = wrapper.find('#toughDataBlock').length
      expect(assertion).toBe(1)
    })

    it('can have an id', () => {
      const wrapper = shallow(baseDataBlockComponent({id: 'dataBlockThing'}))
      const assertion = wrapper.props().id
      expect(assertion).toBe('dataBlockThing')
    })
  })

  describe('dataItem component tests', () => {
    it('matches the snapshot', () => {
      const component = baseDataItemComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseDataItemComponent({id: 'toughDataItem'}))
      const assertion = wrapper.find('#toughDataItem').length
      expect(assertion).toBe(1)
    })

    it('can have an id', () => {
      const wrapper = shallow(baseDataItemComponent({id: 'dataItemThing'}))
      const assertion = wrapper.props().id
      expect(assertion).toBe('dataItemThing')
    })

    it('can have a custom color', () => {
      const wrapper = shallow(baseDataItemComponent({color: 'purple'}))
      const assertion = wrapper.props().color
      expect(assertion).toBe('purple')
    })

    it('can pass a custom counter', () => {
      const wrapper = shallow(baseDataItemComponent({counter: '123123'}))
      const assertion = wrapper.html().includes('123123')
      expect(assertion).toBe(true)
    })

    it('can pass a custom text', () => {
      const wrapper = shallow(baseDataItemComponent({text: 'yolo'}))
      const assertion = wrapper.html().includes('yolo')
      expect(assertion).toBe(true)
    })

    it('can pass a custom description', () => {
      const wrapper = shallow(baseDataItemComponent({description: 'WHAZZUP'}))
      const assertion = wrapper.html().includes('WHAZZUP')
      expect(assertion).toBe(true)
    })
  })
})
