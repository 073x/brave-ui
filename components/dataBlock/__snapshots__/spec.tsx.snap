// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataBlock tests dataBlock component tests matches the snapshot 1`] = `
<ul
  className="sc-bdVaJa rqnMM"
  id={undefined}
  list={undefined}
/>
`;

exports[`dataBlock tests dataItem component tests matches the snapshot 1`] = `
<li
  className="sc-bwzfXH IXpOQ"
  color={undefined}
  id={undefined}
  onClick={undefined}
  size="medium"
>
  <span
    className="sc-htpNat fymEco"
    color={undefined}
    size="medium"
  />
</li>
`;
